posts/2015/03/21/cdl-2015.md

```diff
@@ -0,0 +1,158 @@
+
+For generating the [tests](https://github.com/palcu/convertor/tree/master/teste), I hacked a [Python script](https://github.com/palcu/convertor/blob/master/generator_teste.py) that outputted a JSON. Three tests were special because they had random whitespace. One test was a corner case where there was only one object with lots of keys, and another one had lots of objects with a single key.
+
+The problem gathered 2812 submissions from 158 students.
+
+## Statistics about the online submissions
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/1vo6uj3WJrGRgIV3lGFuwjpSw9BKSIY9Yjby8xWVJD64/pubchart?oid=2010820145&amp;format=interactive"></iframe>
+
+I was expecting more people to fail the problem, but over 50% of the scores are perfect.
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/1vo6uj3WJrGRgIV3lGFuwjpSw9BKSIY9Yjby8xWVJD64/pubchart?oid=116586824&amp;format=interactive"></iframe>
+
+The average number of submissions for a perfect problem was 11. We had a participant who submitted 124 times.
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/1i_n0RWZCyUWUhlI9eonmlL5Dhh_EZhO1Nm0QjF_f3Qs/pubchart?oid=436588329&amp;format=interactive"></iframe>
+
+The most failed test was [number 8](https://github.com/palcu/convertor/blob/master/teste/grader_test8.in). What's special about it is that it contains a single object, and most of the students assume that after the first object ending curly brace, they will have a comma.
+
+A reason why the last 3 tests have the most failing submissions is that they are really big and most of the students preferred to use `getchar` for I/O. Doing that, you overwhelm the operating system with lots of calls. A better approach is to use a buffer.
+
+Also, I've counted the time students spent solving the problem. The average time for any student was 6 hours and 15 minutes, and the average time for a student who successfully passed all the tests was 3 hours and 45 minutes. I made the assumption that if a student submitted a problem for the first time, I added an hour to his solving time, and if the difference between 2 submissions is more than 4 hours, I'm also adding one hour for his next submission. In total, students spent 40 days of time trying to get all the tests passed.
```

Comment:
Remove the "I made the assumption ...to his solving time,", replace it with "To estimate the time spent on solving the problem, I added an hour to the solving time for the first submissions,..."